server/backend/api/app/routers/router_application.py

```diff
@@ -43,3 +49,36 @@ def get_fam_application_user_role_assignment(
                  # of results = {len(app_user_role_assignment)}"
     )
     return app_user_role_assignment
+
+
+@router.get(
+    "/{application_id}/users/{user_id}",
+    response_model=FamUserInfoSchema,
+    status_code=200,
+    dependencies=[Depends(authorize_by_app_id)],
+    summary="Retrieve User Information by User ID under an application",
+    responses={
+        404: {
+            "description": "User not found",
+            "content": {"application/json": {"example": {"detail": "User not found"}}},
+        }
+    },
+)
+async def get_user_by_user_id(
+    user_id: int,
+    db: Session = Depends(database.get_db),
+):
+    """
+    Retrieve the user data for a given user id under an authorized application.
+
+    Args:
+        userId (int): The ID of the user.
+        applicationId (int): The ID of the application the user has access to.
+
+    Returns:
+        FamUserInfoSchema: The user information corresponding to the provided userId.
+    """
+    user = crud_user.get_user(user_id=user_id, db=db)
+    if not user:
+        raise HTTPException(status_code=404, detail="User not found")
```

Comment:
I have different view of 404; but maybe both consider arguably correct or incorrect; even just google for it would not find what is right. I would ok for leaving it.